Improve fastlane Fastfile

janpio opened the pull request:
For #11335, #11445 and #11443 we could use a more flexible `Fastfile`. This PR starts this process by making it a bit more readable by grouping and reordering the lanes and splitting some of them into multiple sub lanes.

janpio committed:
Improve Fastfile:
- Reorder lanes and methods
- Add comments for "grouping"
- Split "validate_repo" in mulitple lanes
- Add missing lane descs

fastlane-bot-helper commented:
<!--
  0 Errors
  1 Warning: Pull Request is Work in Progre...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Pull Request is Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

janpio commented:
The first commit includes:

- Reorder lanes and methods
- Add comments for "grouping"
- Split "validate_repo" in mulitple lanes
- Add missing lane descs

It doesn't really change any code, but this is not very visible here in Github. I locally use WinMerge for diffs, which differentiates between changed or added lines and moved lines. I include screencaptures of a comparison of the before and after state of this commit:

![1](https://user-images.githubusercontent.com/183673/34642536-c9609b1c-f314-11e7-8e40-875b7ebe7668.PNG)
![2](https://user-images.githubusercontent.com/183673/34642537-caf77590-f314-11e7-90a0-e62ebcdbd45c.PNG)
![3](https://user-images.githubusercontent.com/183673/34642538-cc7fce4e-f314-11e7-835a-e4ad1f677b29.PNG)
![4](https://user-images.githubusercontent.com/183673/34642540-ce99fb1e-f314-11e7-9cee-0a3fe972c78d.PNG)
![5](https://user-images.githubusercontent.com/183673/34642542-d037bb6e-f314-11e7-8816-f24cecfc6982.PNG)
![6](https://user-images.githubusercontent.com/183673/34642543-d1fe738e-f314-11e7-850a-767933cc3e6a.PNG)
![7](https://user-images.githubusercontent.com/183673/34642544-d3cb0056-f314-11e7-8ec4-22bd93b9af63.PNG)
![8](https://user-images.githubusercontent.com/183673/34642545-d5bb8a34-f314-11e7-83ca-ffeb36137b51.PNG)
![9](https://user-images.githubusercontent.com/183673/34642546-d74d170a-f314-11e7-9f41-f95f4ac7a79d.PNG)
![10](https://user-images.githubusercontent.com/183673/34642547-daf6a038-f314-11e7-8281-6d3726b2f6b1.PNG)
![11](https://user-images.githubusercontent.com/183673/34642549-dcc7a7fe-f314-11e7-9c99-a99e708971f7.PNG)

- left = old, right = new
- yellow = changed (added, deleted, edited)
- orange-ish = moved
- white unchanged

janpio commented:
Naming of the lanes could probably still be improved: `validate_repo, verify_source, execute_tests,  validate_docs, verify_docs, ` - What exactly is the difference between `validate` and `verify`? Suggestions?
